spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -196,4 +198,39 @@ protected static Object convertConstant(Type type, Object value) {
     }
     return value;
   }
+
+  protected void findMoreCompoundConstants(Schema tableSchema, Map<Integer, ?> idToConstant) {
```

Comment:
Can you please update just one version of Spark and then we'll port the changes to the other versions in separate PRs? That way, each PR is smaller and we don't have to review identical code multiple times.

Reply:
I will only make this pr to spark3.2